include/fixed_containers/struct_view.hpp

```diff
@@ -568,38 +568,51 @@ class StructView
     }
 
     template <typename S>
-    void add_path(S&& instance, const PathNameChain& path)
+    bool add_path(S&& instance, const PathNameChain& path)
     {
         auto path_properties_map = extract_path_properties_of_filtered<S, 1, 1>(
             std::forward<S>(instance), std::optional<FixedSet<PathNameChain, 1>>({path}));
-        auto [_, was_inserted] = path_properties_.try_emplace(path, path_properties_map.at(path));
-        assert_or_abort(was_inserted);
+
+        const auto itr = path_properties_map.find(path);
+        if (itr == path_properties_map.cend())
+        {
+            return false;
+        }
+
+        path_properties_.emplace(path, itr->second);
+        return true;
     }
 
     template <typename S>
-    void add_path(const PathNameChain& path)
+    bool add_path(const PathNameChain& path)
     {
         S instance{};
-        add_path(instance, path);
+        return add_path(instance, path);
     }
 
     template <typename S, typename PathSet>
-    void add_paths(S&& instance, const PathSet& paths)
+    bool add_paths(S&& instance, const PathSet& paths)
     {
         auto path_properties_map =
             extract_path_properties_of_filtered(std::forward<S>(instance), paths);
+        if (path_properties_map.size() != paths.size())
+        {
+            return false;
+        }
+
         for (const auto& [path, path_properties] : path_properties_map)
         {
-            auto [_, was_inserted] = path_properties_.try_emplace(path, path_properties);
-            assert_or_abort(was_inserted);
+            path_properties_.emplace(path, path_properties);
         }
+
+        return true;
     }
 
     template <typename S, typename PathSet>
-    void add_paths(const PathSet& paths)
+    bool add_paths(const PathSet& paths)
     {
         S instance{};
```

Comment:
Allocating a templated type on the stack can lead to stackoverflow since the size of S is unbounded. I am curious, is this our "responsibility/fault" if someone gives us a big S and we crash? I wonder how other generic code deals with this.

For context, when debugging a crash related to `struct_view` with @alexkaratarakis a few days ago , we initially suspected it was due to this line.

Reply:
Theoretically only the type information is required in order to traverse the struct and construct path properties. I'm not completely sure why we had to allocate an instance of the object here.

Unfortunately despite the fact that memory is never read from the struct instance, the compiler does not optimize out the stack allocation for the default value.

I tried it by adding a really large field to the unit test's `FlatSuperStruct1` definition.

```diff
--- a/test/struct_view_test.cpp
+++ b/test/struct_view_test.cpp
@@ -49,6 +49,7 @@ struct FlatSuperStruct1
     std::int32_t ignore2{};
     std::int32_t retain2{};
     std::int16_t ignore3{};
+    unsigned char bigbuf[1024*1024];
 };

 struct FlatSubStruct1
```

Even compiling with `-O3` the large allocation is still present in the disassembly.

```
$ sym_name="_ZN16fixed_containers11struct_view30StructView_StructViewFlat_Test8TestBodyEv"
$ echo $sym_name | c++filt
fixed_containers::struct_view::StructView_StructViewFlat_Test::TestBody()
$ objdump --disassemble=$sym_name ./struct_view_test | head -n22
...
000000000013e780 <_ZN16fixed_containers11struct_view30StructView_StructViewFlat_Test8TestBodyEv>:
  13e780:       55                      push   %rbp
  13e781:       48 89 e5                mov    %rsp,%rbp
  13e784:       41 57                   push   %r15
  13e786:       41 56                   push   %r14
  13e788:       41 55                   push   %r13
  13e78a:       41 54                   push   %r12
  13e78c:       53                      push   %rbx
  13e78d:       48 83 e4 e0             and    $0xffffffffffffffe0,%rsp
  13e791:       48 81 ec 80 c4 10 00    sub    $0x10c480,%rsp
```

Reply:
Okay, this can be outside of this PR. I am curious what you think about this idea:
Instead of doing the allocation inside `get_path`, how about we change the api to require passing in an object of the type of interest? The STL does something similar with `std::accumulate`. This way, the caller can allocate the object however they want and we do not have to worry about causing a stack-overflow?

Reply:
That sounds like a good idea. We can address that in a follow-up.